python/ray/dag/compiled_dag_node.py

```diff
@@ -1724,6 +1748,9 @@ def teardown(self, wait: bool):
                     self.wait_teardown()
                     logger.info("Teardown complete")
 
+                with self.in_teardown_lock:
+                    self._teardown_done = True
```

Comment:
When `wait` is false, we directly mark `_teardown_done` as true without waiting? This seems problematic.

Now that `wait` is always true, can we clean up the `wait` parameter and logic?